hack/validate-migration.sh

```diff
@@ -0,0 +1,332 @@
+#!/usr/bin/env bash
+
+# Validate migration file introduced by a branch.
+#
+# This script can be run in the CI against a PR or in local from a topic branch.
+# Before run, all local changes have to be committed.
+#
+# Network is required to execute this script.
+#
+# Checks are implemented as functions whose name has prefix `check_`. Each of
+# them exits with status code 0 to indicate pass, otherwise exits
+# script execution immediately with non-zero code.
+#
+# Test with a local cluster:
+#
+# - Before starting test with local cluster, launch one by `kind create
+#   cluster'.  Then, install tekton pipelines
+#   https://tekton.dev/docs/pipelines/install/#installation
+#
+# - Set IN_CLUSTER to true.
+
+set -euo pipefail
+
+SCRIPTDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+cd "$SCRIPTDIR/.." || exit 1
+unset SCRIPTDIR
+
+declare -r BUILD_PIPELINE_CONFIG=https://raw.githubusercontent.com/redhat-appstudio/infra-deployments/refs/heads/main/components/build-service/base/build-pipeline-config/build-pipeline-config.yaml
```

Comment:
In the commit message, could you describe more about what the check does and why? In particular, I don't quite understand why we need the pipeline bundles from redhat-appstudio/infra-deployments